objectcomputing/check-ins
# Upgrade to Gradle 8.8 and cleanup build

This PR upgrades Gradle to 8.8 and performs some cleanup, namely:

1. Removes unused configurations
2. [Follows current best practices](https://docs.gradle.org/current/userguide/cross_project_publications.html#sec:variant-aware-sharing) to depend on the web-ui yarnBuild output
3. Fixes the publication task (I think)... At least `./gradlew publishToMavenLocal` seems correct now 🤔 
4. Simplifies the main build file by removing subproject configuration
5. Removes the idea plugin which I don't believe is required any more (as import works)
6. The sonar plugin didn't support Gradle 8, so this PR updates it to latest, and configures it to work.

### Running the sonar plugin to test it

- Run sonar locally with `docker run -d --name sonarqube -p 9000:9000 sonarqube` then visit localhost:9000
- When it's running log in as an admin (with admin/admin) and then change your password as requested
- Create a local project and give it a name and a key `check-ins`
- On the next screen, select "Use the global setting" and click "Create project"
- Select "Locally" for the analysis method
- Click "Generate" to create a token, then click "continue"
- Select "gradle", copy the command, and run it locally in a terminal window

Closes #2461
